### PR TITLE
Add Shape inference

### LINKED ISSUE #39: [FEATURE] Add shape inference to onnx export, for TensorRT runtime compatibility
**Is your feature request related to a problem? Please describe.**
Yes, currently the onnx model cannot be used on onnxruntime with tensorrt backend

**Describe the solution you'd like**
Adapt script to add shape inference derived from https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/core/providers/nuphar/scripts/symbolic_shape_infer.py into our exporter

### PR DESCRIPTION
**Pull request by alifahrri**
## Type of changes

 

Please check the type of change your PR introduces:
- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?
no shape inference

- close #39 

Related issues : 

## What is the new behavior?

- add shape inference
- add tensorrt docker
- add model explorer

## Checklist

- [ ] Unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have filled the `[Unreleased]` [Changelog](https://github.com/nodefluxio/vortex/blob/master/CHANGELOG.md)

### CODE REVIEW

**Comment by alifahrri** [comment]
Note:
- the tensorrt docker is adapted from github.com/microsoft/onnxruntime/, and use python3.7
- for now only vortex runtime available from tensorrt docker
- known dependency issues: albumentations, enforce
- model explorer provided also measure inference time, provided as alternative to model reporting

File: scripts/model_explorer.py [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+                runtime=self.runtime_var.get(),
+            )
+            print('loading model, this may take a while')
+            self.model = InferenceHelper.create_runtime_model(**kwargs)
+            print('loading model done:', type(self.model.model))
+            self.runtime = []
+            title = f"{self.title}: {self.model_path}"
+            self.root.title(title)
+        else:
+            print(f'{filename} doesnt exists')
```
Comment (triwahyuu) [inline scripts/model_explorer.py:167]: `filename` is not defined here, where is this variable come from?

**Review by triwahyuu — COMMENTED** [review]
can you elaborate what the `scripts/model_explorer.py` does? I tried to use it but I don't understand how to do anything with it.

**Comment by triwahyuu** [comment]
I think you forgot to add `sympy` in the requirements

**Comment by triwahyuu** [comment]
I have fixed the tests fail in #85

File: scripts/model_explorer.py [hunk 2]
```diff
@@ -0,0 +1,247 @@
+import tkinter
+import numpy as np
+from tkinter import ttk
+from tkinter import filedialog
+
+from pathlib import Path
+import sys
+
+# get inference helper from runtime eample
+root = Path(__file__).parent.parent
```
[the remark this answers is not in the document]
Comment (alifahrri) [reply scripts/model_explorer.py:10]: apparently it doesnt work
```
root = Path(__file__).parent[1]
TypeError: 'PosixPath' object does not support indexing`
```

[the remark this answers is not in the document]
Comment (alifahrri) [reply scripts/model_explorer.py]: fixed

File: scripts/model_explorer.py (hunk 1 above)
Comment (alifahrri) [reply scripts/model_explorer.py]: fixed

**Comment by alifahrri** [comment]
> I think you forgot to add `sympy` in the requirements



> I think you forgot to add `sympy` in the requirements

fixed

File: scripts/model_explorer.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (alifahrri) [reply scripts/model_explorer.py]: model_explorer is similar to dataset visualizer but intended for trying model inference interactively

**Comment by alifahrri** [comment]
> can you elaborate what the `scripts/model_explorer.py` does? I tried to use it but I don't understand how to do anything with it.

model_explorer is similar to dataset visualizer but intended for trying model inference interactively

**Comment by triwahyuu** [comment]
how can I use it? after I load the model, I dont know what to do

**Comment by triwahyuu** [comment]
ahh, I got it now, all good. LGTM.

**Review by triwahyuu — APPROVED** [review]

**Pull request merged**